.github/workflows/release.yaml

```diff
@@ -0,0 +1,26 @@
+name: Publish to PyPI
+on:
+  push:
+    tags:
+      - "*.*.*"
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    if: github.repository == 'dask/dask-jobqueue'
+    environment:
+      name: pypi
+      url: https://pypi.org/p/dask-jobqueue
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.10"
+      - name: Build package
+        run: pip install build && pyproject-build
+      - name: Publish
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
For anyone interested in how this works without passing any credentials I had to log into PyPI and set up GitHub Actions as a [trusted publisher](https://docs.pypi.org/trusted-publishers/adding-a-publisher/).